presto-spi/src/main/java/io/prestosql/spi/connector/ProjectionApplicationResult.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.connector;
+
+import io.prestosql.spi.expression.ConnectorExpression;
+import io.prestosql.spi.type.Type;
+
+import java.util.List;
+
+public class ProjectionApplicationResult<T>
```

Comment:
Can projection pushdown consume projections partially?
For example: `a = b + c + pow(d)` -> `a = e + pow(d)`?

Reply:
Eventually, yes. But not in this initial prototype, as the only thing that’s currently supported in the mini expression language are simple column references